### PR TITLE
Expect less chaos: use explicit call indices

### PR DESCRIPTION
**Pull request by gilescope**
Set call index explicitly so that calls do not inadvertently change their index in the pallet.

### CODE REVIEW

**Review by joepetrowski — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]
Looks like only `test/runtime/src/test_pallet.rs` is missing.  
You can also put it into dev-mode, since otherwise there will be a compiler warning in future Substrate versions https://github.com/paritytech/substrate/pull/12894

**Review by muharem — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2178121

File: test/runtime/src/test_pallet.rs [hunk 1]
```diff
@@ -17,6 +17,7 @@
 /// A special pallet that exposes dispatchables that are only useful for testing.
 pub use pallet::*;
 
+#[pallet(dev_mode)]
```
Comment (ggwpez) [inline test/runtime/src/test_pallet.rs:20]: AFAIK the syntax is `#[frame_support::pallet(dev_mode)]`.

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**